src/test/java/com/greenfiling/smclient/SupplierCostClient_UnitTest.java

```diff
@@ -0,0 +1,70 @@
+package com.greenfiling.smclient;
+
+import static org.hamcrest.CoreMatchers.notNullValue;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import com.greenfiling.smclient.internal.JsonHandle;
+import com.greenfiling.smclient.model.SupplierCost;
+import com.greenfiling.smclient.model.exchange.Show;
+import com.greenfiling.smclient.model.exchange.SupplierCostFilter;
+import com.greenfiling.smclient.model.exchange.SupplierCostFilter.JobType;
+import com.greenfiling.smclient.model.exchange.SupplierCostFilter.ServiceLevel;
+
+public class SupplierCostClient_UnitTest {
+  private static ApiHandle apiHandle = null;
+  private static SupplierCostClient client = null;
+
+  @BeforeClass
+  public static void setUpClass() {
+    apiHandle = new ApiHandle.Builder().apiKey(TestHelper.VALID_API_KEY).apiEndpoint(ApiHandle.DEFAULT_ENDPOINT_BASE).build();
+    client = new SupplierCostClient(apiHandle);
+  }
+
+  @Test
+  public void testShowCostByZip_HappyPath() throws Exception {
+    Show<SupplierCost> response = client.show(47374);
+
+    assertThat(response.getData().getAmount(), notNullValue());
+
+    System.out.println("amount = " + response.getData().getAmount());
+    System.out.println("updated_at = " + response.getData().getUpdatedAt());
+    System.out.println("created_at = " + response.getData().getCreatedAt());
+
+    System.out.println("re-serialized: " + JsonHandle.get().getGsonWithNulls().toJson(response));
+  }
+
+  @Test
+  public void testShowCostByZip_WithQueryParameters_HappyPath() throws Exception {
+    SupplierCostFilter request = new SupplierCostFilter();
+    request.setZipCode(47374);
+    request.setJobType(JobType.CCD);
+    request.setServiceLevel(ServiceLevel.ROUTINE);
+
+    Show<SupplierCost> response = client.show(request);
+
+    assertThat(response.getData().getAmount(), notNullValue());
+
```

Comment:
Great idea. In doing so, the j-tests revealed a bleed-over bug (yikes) in that SupplierCostFilter can't be a property of a class but should be local to the show-call only in case the client is being re-used for multiple calls. I still need to address this bug

Reply:
bleed-over resolved in commit c8bd40c..5cb07a0 . Was avoiding changing too much ApiClient at the beginning but expending show implementation to allow optional filtering seems like the right thing to do.

Reply:
It's a bummer that you ended up touching more core code than we thought might be necessary to implement this non-standard thing (filters on a show interface), but I kind of anticipated it.  The assert change and the refactor it prompted look good